### ISSUE TITLE
Generic type with `float` bound generates `unsupported-operator` error

### ISSUE THREAD

**Issue by sin-ack** [astral-sh/ty#2652]
### Summary

Playground link: https://play.ty.dev/5b750751-87df-4f7f-9aea-5ccaeabf5f15

Code:

```py
from typing import Generic, TypeVar

T = TypeVar("T", bound=float)


class Range(Generic[T]):
    min: T
    max: T

    def __init__(self, min: T, max: T) -> None:
        self.min = min
        self.max = max

    def __contains__(self, value: T) -> bool:
        return self.min <= value <= self.max
```

ty does not recognize that the generic type has a bound on `float` (which has `__le__`), and complains with ``Unsupported `<=` operation``. 

Both Pyright (1.1.408) and mypy (1.19.1) find no issues with this code.

### Version

0.0.14

**Comment by carljm** [comment]
Thanks for the report!

It works fine with a bound of `int`, so the issue seems specific to a typevar with an upper bound which is a union (since the type expression `float` means `int | float`). It also [works fine](https://play.ty.dev/ad15e8fd-42d4-4b24-aead-a95362b98f93) with a direct type of `float`, so the issue is not general to this union, it only manifests with a typevar upper bound.

**Comment by sin-ack** [comment]
> so the issue seems specific to a typevar with an upper bound which is a union (since the type expression float means int | float).

Why would that be the case? `int` and `float` share no common ancestor, no?

**Comment by carljm** [comment]
> > so the issue seems specific to a typevar with an upper bound which is a union (since the type expression float means int | float).
> 
> Why would that be the case? `int` and `float` share no common ancestor, no?

Correct. But (unfortunately?) the Python typing specification has always contained an unclear mandate that type-checkers should allow passing an `int` where a `float` is expected, even though `int` is not a subclass of `float`, and lots of code relies on this. Type checkers implement this unclear mandate in various ways, most of them unsound. ty chooses a sound interpretation, where a type annotation of `float` means `int | float`. This union is meaningful precisely because `int` and (real) `float` are unrelated types.
